classes/services/PKPSubmissionFileService.inc.php

```diff
@@ -0,0 +1,837 @@
+					 // IF no prior notification exists
+					 // OR if editor has logged in after the last revision upload
+					 // OR the last upload and notification was sent more than a day ago,
+					 // THEN send a new notification
+					if (is_null($lastNotification) || strtotime($editor->getDateLastLogin()) > $lastNotification || strtotime('-1 day') > $lastNotification){
+						$mail->addRecipient($editor->getEmail(), $editor->getFullName());
+					}
+				}
+				// Get uploader name
+				$mail->assignParams(array(
+					'authorName' => $uploader->getFullName(),
+					'editorialContactSignature' => $context->getData('contactName'),
+					'submissionUrl' => $request->getDispatcher()->url(
+						$request,
+						ROUTE_PAGE,
+						null,
+						'workflow',
+						'index',
+						[
+							$submission->getId(),
+							$reviewRound->getStageId(),
+						]
+					),
+				));
+
+				if ($mail->getRecipients()){
+					if (!$mail->send($request)) {
+						import('classes.notification.NotificationManager');
+						$notificationMgr = new \NotificationManager();
+						$notificationMgr->createTrivialNotification($request->getUser()->getId(), NOTIFICATION_TYPE_ERROR, ['contents' => __('email.compose.error')]);
+					}
+				}
+			}
+		}
+
+		HookRegistry::call('SubmissionFile::add', [$submissionFile, $request]);
+
+		return $submissionFile;
+	}
+
+	/**
+	 * @copydoc \PKP\Services\EntityProperties\EntityWriteInterface::edit()
+	 */
+	public function edit($submissionFile, $params, $request) {
+		$newFileUploaded = !empty($params['fileId']) && $params['fileId'] !== $submissionFile->getData('fileId');
+		$submissionFile->_data = array_merge($submissionFile->_data, $params);
+		$submissionFile->setData('updatedAt', Core::getCurrentDate());
+
+		HookRegistry::call('SubmissionFile::edit', [$submissionFile, $submissionFile, $params, $request]);
+
+		DAORegistry::getDAO('SubmissionFileDAO')->updateObject($submissionFile);
+
+		import('lib.pkp.classes.log.SubmissionFileLog');
+		import('lib.pkp.classes.log.SubmissionFileEventLogEntry'); // constants
+		\SubmissionFileLog::logEvent(
+			$request,
+			$submissionFile,
+			$newFileUploaded ? SUBMISSION_LOG_FILE_REVISION_UPLOAD : SUBMISSION_LOG_FILE_EDIT,
+			$newFileUploaded ? 'submission.event.revisionUploaded' : 'submission.event.fileEdited',
+			[
+				'fileStage' => $submissionFile->getData('fileStage'),
+				'sourceSubmissionFileId' => $submissionFile->getData('sourceSubmissionFileId'),
+				'submissionFileId' => $submissionFile->getId(),
+				'fileId' => $submissionFile->getData('fileId'),
+				'submissionId' => $submissionFile->getData('submissionId'),
+				'originalFileName' => $submissionFile->getLocalizedData('name'),
+				'username' => $request->getUser()->getUsername(),
+			]
+		);
+
+		import('lib.pkp.classes.log.SubmissionLog');
+		import('classes.log.SubmissionEventLogEntry');
+		$user = $request->getUser();
+		$submission = Services::get('submission')->get($submissionFile->getData('submissionId'));
```

Comment:
Double-logged again.

Reply:
Oh, I see, we're logging in both the submission and submission file logs. NVM!

Reply:
I did the same thing when moving the code over! :joy: